server/src/main/java/org/elasticsearch/search/suggest/completion/CompletionSuggestionBuilder.java

```diff
@@ -298,16 +304,19 @@ public SuggestionContext build(QueryShardContext context) throws IOException {
         if (mappedFieldType == null || mappedFieldType instanceof CompletionFieldMapper.CompletionFieldType == false) {
             throw new IllegalArgumentException("Field [" + suggestionContext.getField() + "] is not a completion suggest field");
         }
-        if (mappedFieldType instanceof CompletionFieldMapper.CompletionFieldType) {
-            CompletionFieldMapper.CompletionFieldType type = (CompletionFieldMapper.CompletionFieldType) mappedFieldType;
-            suggestionContext.setFieldType(type);
-            if (type.hasContextMappings() && contextBytes != null) {
+        CompletionFieldMapper.CompletionFieldType type = (CompletionFieldMapper.CompletionFieldType) mappedFieldType;
+        suggestionContext.setFieldType(type);
+        if (type.hasContextMappings()) {
+            if (contextBytes == null) {
```

Comment:
This checks for a missing `contexts` element in the query, but I think it will miss cases like `"contexts" : {}` or `"contexts" :  { "someField : [] }` which should also emit a deprecation warning. Not sure if this means there should be another check after the queryContexts map is parsed.

Reply:
Good catch, thanks ! This also made me realize that we should deprecate the index side if no contexts are provided. This is needed because these suggestions are not retrievable anymore if we remove the ability to search with no context. I pushed https://github.com/elastic/elasticsearch/pull/30712/commits/f5edec43302eafc2edfea583b93e0b0996e5dc9a to handle all cases of empty context in the query side and to also deprecate indexing without contexts.